### DISCUSSION TITLE
More information on reactivity scope (and the lack of it)

### DISCUSSION THREAD

**Discussion by morgs32** [maxnowack/signaldb#844] [General]
I'm still unclear on what reactivity scope is (there's not a lot to find from googling it either). Perhaps someone could provide an example of an implementation that causes a memory leak BECAUSE we don't have that capability (say with the preact adapter?).

Thanks!

**Comment by maxnowack** [comment]
You are in a reactive scope when the cursor was created in the `effect` function of your reactivity library.
Sorry, the documentation wasn't clear about that. I've made some improvements in https://github.com/maxnowack/signaldb/commit/8192ae6464370a8bc018f7fa59b2013c66783fdb.
See https://signaldb.js.org/reactivity/#reactivity-libraries and https://signaldb.js.org/reactivity/other/#isinscope-dependency-dependency-boolean.

An example of an implementation that causes a memory leak is when call the `<collection>.find()` method outside a reactive scope and don't turn off reactivity in the options (`{ reactive: false }`, this would happen automatically if the adapter supports scope checking). If you're not turning off reactivity manually, SignalDB setups reactivity unnecessarily and is also not able to cleanup this automatically. This also wasn't clearly explained in the documentation and I improved it a bit (see https://signaldb.js.org/reactivity/preact-signals/#adapter).

Besides scope checking you have to do you cleanup of the cursors manually on some libraries (Preact for example). I also improved the docs on this (see https://signaldb.js.org/reactivity/other/#ondispose-callback-void-dependency-dependency or the specific adapter pages).

Just let me know if something is still unclear 🙂

**Reply by morgs32** [reply]
Much clearer now thank you for the explanation and your additions to the docs!
